openedx/features/enterprise_support/api.py

```diff
@@ -473,16 +473,18 @@ def consent_needed_for_course(request, user, course_id, enrollment_exists=False)
     if not enterprise_learner_details:
         consent_needed = False
     else:
+        currently_active_enterprise = enterprise_customer_uuid_for_request(request)
```

Comment:
Wouldn't the following method be more suitable here?
https://github.com/edx/edx-platform/pull/24291/files#diff-d27e76023a81690ef221162664413e70R448

Reply:
We don't need the whole model, just the uuid to filter out the non-relevant enterprise learner details